drivers/kernel_test: rename the kernel test driver

deepaksrma opened the pull request:
- rename the kernel_test_drv as os_test_drv because kernel_test contains kernel and network TCs and also may contain filesystem TCs.
- remove unnecessary header files

Signed-off-by: Deepak sharma <deepak.sh@samsung.com>

Commit:
drivers/kernel_test: rename the kernel test driver

- rename the kernel_test_drv as os_test_drv because kernel_test contains kernel and network TCs and also may contain filesystem TCs.
- remove unnecessary header files

Signed-off-by: Deepak sharma <deepak.sh@samsung.com>

kishore-sn commented on os/drivers/kernel_test/Make.defs:
```diff
@@ -16,10 +16,10 @@
 include kernel_test$(DELIM)kernel$(DELIM)Make.defs
 include kernel_test$(DELIM)network$(DELIM)Make.defs
-CSRCS += kernel_test_drv.c
+CSRCS += os_test_drv.c
```
Maybe better to call it os_api_test_drv.c

deepaksrma replied:
done

kishore-sn commented on os/include/tinyara/kernel_test_drv.h:
```diff
@@ -30,14 +30,14 @@
 /* Configuration
  *
- * CONFIG_DRIVERS_KERNEL_TEST - Enables kernel_test driver support
+ * CONFIG_OS_KERNEL_TEST - Enables kernel_test driver support
```
Pls change comment description also

deepaksrma replied:
done

kishore-sn commented on os/include/tinyara/kernel_test_drv.h:
```diff
@@ -30,14 +30,14 @@
 #include <tinyara/fs/ioctl.h>
 
-#ifdef CONFIG_DRIVERS_KERNEL_TEST
+#ifdef CONFIG_OS_KERNEL_TEST
```
We will also need to rename this header file as os_api_test_drv.h

deepaksrma replied:
done

kishore-sn commented on build/configs/artik053/STDK/defconfig:
```diff
@@ -436,7 +436,7 @@ CONFIG_ADC=y
 CONFIG_ADC_NPOLLWAITERS=2
 # CONFIG_DAC is not set
-# CONFIG_DRIVERS_KERNEL_TEST is not set
+# CONFIG_OS_KERNEL_TEST is not set
```
CONFIG_OS_KERNEL_TEST still refers to kernel test. 
Better to change it as CONFIG_OS_API_TEST

deepaksrma replied:
done

kishore-sn commented on os/drivers/kernel_test/os_test_drv.c:
```diff
@@ -27,8 +27,7 @@
 #include <tinyara/sched.h>
-#include "signal/signal.h"
-#include "kernel_test_proto.h"
+#include "os_test_proto.h"
```
Also, need to rename all api's and data structures.

deepaksrma replied:
done

Commit:
configs/defconfig: modify the defconfig

- replace CONFIG_DRIVERS_KERNEL_TEST with CONFIG_DRIVERS_OS_API_TEST. because in os_api_test, we test kernel, network and fs apis.

Signed-off-by: Deepak sharma <deepak.sh@samsung.com>

Commit:
testcase/network/tc_net_pbuf.c: fix coding violations rule

Signed-off-by: Deepak sharma <deepak.sh@samsung.com>
